lib/sgtn-client/api/translation.rb

```diff
@@ -11,42 +11,39 @@ module Core
   end
 
   module Translation
-    def self.getString(component, key, locale)
-      SgtnClient.logger.debug "[Translation.getString]component: #{component}, key: #{key}, locale: #{locale}"
+    def self.getString(component, key, locale, *args, **kwargs)
+      SgtnClient.logger.debug "[Translation.getString]component: #{component}, key: #{key}, locale: #{locale}, args: #{args}"
+      locale = LocaleUtil.get_best_locale(locale)
       str = getTranslation(component, key, locale)
-      if str.nil? && !LocaleUtil.is_source_locale(locale)
-        str = getTranslation(component, key, LocaleUtil.get_source_locale)
+      if str
+        locale = str.locale if str.is_a?(SgtnClient::StringUtil)
+      elsif !LocaleUtil.is_source_locale(locale)
+        locale = LocaleUtil.get_source_locale
+        str = getTranslation(component, key, locale)
+      end
+      return if str.nil?
+
+      if !kwargs.empty?
+        str.localize(locale) % kwargs
+      elsif !args.empty?
+        str.localize(locale) % args
+      else
+        str
       end
-      str
     end
 
     def self.getString_p(component, key, plural_args, locale)
```

Comment:
Is a separate getString_p necessary? Why not call getString directly? So instead of
`SgtnClient::Translation.getString_p("JAVA", "plural_key", { :cat_count => 1 }, "zh-Hans")`, 
call `SgtnClient::Translation.getString("JAVA", "plural_key", "zh-Hans", { :cat_count => 1 })`

Reply:
This is for compatibility. Can consider to remove them.

Reply:
If only for compatibility with existing code in CHT, we can mark this as deprecated and suggest to use get_string. We can do this in another PR if you like.